### PR TITLE
forcedialog: integrate children of nested fields

### PR DESCRIPTION
**Pull request by yetanotherion**
The current implementation gather the fields of ui elements
that are of type nested (NestedParameter).
However, some UI elements might inherit from NestedParameter, but still
have another type. This commit permits to handle such, but
ducktyping nested ui fields, on the presence of the
attribute fields.

Change-Id: I74f3a639ea74da0baae7ce6038ab91e8d925255a

### CODE REVIEW

**Comment by tardyp** [comment]
please remove change-id from the commit message. I like your design of having a custom nested parameter. I think this would be a good idea to publish your recipe.

**Comment by yetanotherion** [comment]
@tardyp the commit message has been cleaned. I'll work on upstreaming an example
this week end, meanwhile do we agree on merging this one as it is ?

**Comment by tardyp** [comment]
👍

Le jeu. 21 mai 2015 10:00, yetanotherion <notifications@github.com> a
écrit :

**Comment by yetanotherion** [comment]
the test failed in
which does not see to be related to that commit, @tardyp do you think we have 
another racy issue in the trigger ?

===============================================================================
[ERROR]
Traceback (most recent call last):
Failure: twisted.internet.defer.TimeoutError: <buildbot.test.integration.test_trigger.TriggeringMaster testMethod=test_trigger> (test_trigger) still running at 120.0 secs
buildbot.test.integration.test_trigger.TriggeringMaster.test_trigger
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/twisted/internet/defer.py", line 1070, in _inlineCallbacks
    result = g.send(result)
  File "/home/travis/build/buildbot/buildbot/master/buildbot/test/util/integration.py", line 128, in tearDown
    raise self.failureException(dump.getvalue())
twisted.trial.unittest.FailTest: FAILED! dumping build db for debug
buildbot.test.integration.test_trigger.TriggeringMaster.test_trigger

**Comment by tardyp** [comment]
We still have the race condition you found in your other PR. No need to bother restarting travis, as other tests are ok

**Comment by yetanotherion** [comment]
@sa2ajj @tardyp the example is now available. If the travis test fails note that https://github.com/buildbot/buildbot/pull/1678 might have corrected it.

File: www/nestedexample/README [hunk 1]
```diff
@@ -0,0 +1,25 @@
+This plugin permits to create two linked UI elements that can be integrated
```
Comment (tardyp) [inline www/nestedexample/README]: I think this is not clear how the UI element are linked. Please explicit a little bit more.

File: www/nestedexample/buildbot_nestedexample/__init__.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+class NestedExample(NestedParameter):
+
+    """UI zone"""
+    type = "nestedexample"
+    PIZZA = "pizza"
+    INGREDIENTS = "ingredients"
+
+    def __init__(self, **kw):
+        pizzaInput = StringParameter(label="type the name of your pizza",
```
Comment (tardyp) [inline www/nestedexample/buildbot_nestedexample/__init__.py]: I like the example!

**Comment by yetanotherion** [comment]
@tardyp README updated

**Comment by yetanotherion** [comment]
this failure is new, but I don't see how it can be related to the code in that PR

Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/twisted/internet/defer.py", line 1070, in _inlineCallbacks
    result = g.send(result)
  File "/home/travis/build/buildbot/buildbot/master/buildbot/test/util/integration.py", line 128, in tearDown
    raise self.failureException(dump.getvalue())
twisted.trial.unittest.FailTest: FAILED! dumping build db for debug
buildbot.test.integration.test_mailnotifier.MailMaster.test_notifiy_for_build

**Comment by yetanotherion** [comment]
@tardyp README updated

**Comment by yetanotherion** [comment]
@sa2ajj @tardyp are we ok to merge it ?

**Comment by tardyp** [comment]
Ok for me

Le sam. 23 mai 2015 07:44, yetanotherion <notifications@github.com> a
écrit :

**Comment by yetanotherion** [comment]
thx :)

2015-05-24 13:14 GMT+02:00 Mikhail Sobolev <notifications@github.com>:

> Merged #1683 <https://github.com/buildbot/buildbot/pull/1683>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1683#event-312537763>.
>



-- 
Ion

**Pull request merged**